### DISCUSSION TITLE
eslint plugin: add a rule that prevents unnecessary action decorator

### DISCUSSION THREAD

**Discussion by todor-a** [mobxjs/mobx#3951] [Ideas]
Just fixed a bug that was caused by an unnecessary action.bound - no state was being modified inside the function. Instead, it was acting like a getter that depended on an observable in the store. Subscribers were not alerted when this observable was changed. 

I was wondering if it makes sense to have an eslint rule for that?

**Comment by mweststrate** [comment]
It's hard to have linters tell you that mismatches between intent and what
someone is literally declaring? Such a lint rule would be really
complicated. E.g. if the actions a.x() or z(a), where a is an observable
object, you cannot tell at a glance whether x resp z will mutate or compute.

However theoretically we could throw if an action is called from a reactive
reading context. (There might be an actual flag for that, not sure from top
of my head)
